server/lib/payments.js

```diff
@@ -914,11 +914,15 @@ export const getHostFeeSharePercent = async (order, host = null) => {
   const possibleValues = [];
 
   if (order) {
-    if (order.paymentMethod?.service === 'stripe' && order.paymentMethod?.type === 'creditcard') {
-      possibleValues.push(plan?.creditCardHostFeeSharePercent);
+    // Preload payment method
+    if (!order.paymentMethod && order.PaymentMethodId) {
```

Comment:
Maybe simpler like this. But that's a detail. ```suggestion
    if (!order.paymentMethod) {
```